### ISSUE TITLE
Previously-unseen warning with EVReflection v2.13.1, EVCloudKitDao v2.14.2 and CKRecordID

### ISSUE THREAD

**Issue by tbaggett** [evermeer/EVCloudKitDao#39]
I am seeing this warning every time a CKRecordID instance is handled:

    WARNING: Should have been handled somewere else: value <CKRecordID: 0x127f971b0; 02F113F2-39F5-4E21-83A0-008CFB786440:(_defaultZone:__defaultOwner__)> type CKRecordID

Note that I am compiling things for tvOS 9.1. I am also restricting EVCloudKitDao and EVReflection to extension-compatible APIs so they can be used with my tvOS top shelf extension.

Can this be safely ignored?

**Comment by tbaggett** [comment]
And then there this monster appears as well:

    WARNING: Should have been handled somewere else: value <62706c69 73743030 d4010203 0405063a 3b582476 65727369 6f6e5824 6f626a65 63747359 24617263 68697665 72542474 6f701200 0186a0af 10120708 09101117 18192024 282b2e32 33373839 55246e75 6c6c5b50 6572666f 726d616e 6365d30a 0b0c0d0e 0f562463 6c617373 5a526563 6f72644e 616d6556 5a6f6e65 49448008 80038004 5f102446 36423830 3445332d 42394534 2d343332 352d3937 30312d42 31314644 34394631 363733d3 12130a14 1516585a 6f6e654e 616d6559 6f776e65 724e616d 65800580 0680075c 5f646566 61756c74 5a6f6e65 5f10105f 5f646566 61756c74 4f776e65 725f5fd2 1a1b1c1d 5a24636c 6173736e 616d6558 24636c61 73736573 5e434b52 65636f72 645a6f6e 654944a2 1e1f5e43 4b526563 6f72645a 6f6e6549 44584e53 4f626a65 6374d21a 1b21225a 434b5265 636f7264 4944a223 1f5a434b 5265636f 72644944 d2250a26 27574e53 2e74696d 652341bc 13ddc17e 353f800a d21a1b29 2a564e53 44617465 a2291fd2 250a2c27 2341bc13 ddc17e35 3f800ad3 0a0b0c0d 300f8008 800d8004 5f10105f 5f646566 61756c74 4f776e65 725f5fd3 0a0b0c0d 350f8008 800f8004 5f10105f 5f646566 61756c74 4f776e65 725f5f58 69687479 3677676c 58417070 6c652054 565f100f 4e534b65 79656441 72636869 766572df 10153c3d 3e3f4041 42434445 46474849 4a4b4c4d 4e4f5051 52535455 56555855 5a5b515d 5e5f5555 55555555 59576173 43616368 65645a50 65726d69 7373696f 6e5a5265 636f7264 54797065 5d4b6e6f 776e546f 53657276 65725b46 616b6553 68617265 49445852 65636f72 6449445f 10165a6f 6e655072 6f746563 74696f6e 44617461 45746167 54455461 675f100f 50726576 696f7573 53686172 6549445f 10184c61 73744d6f 64696669 65645573 65725265 636f7264 49445f10 13437265 61746f72 55736572 5265636f 72644944 5f101148 61735570 64617465 64536861 72654944 5b526563 6f72644d 74696d65 5b526563 6f726443 74696d65 5f10104d 6f646966 69656442 79446576 6963655e 50726f74 65637469 6f6e4461 74615f10 17536861 72655072 6f746563 74696f6e 44617461 45746167 57536861 72654944 5f101243 6f6e666c 6963744c 6f736572 45746167 735f1012 50726f74 65637469 6f6e4461 74614574 61675f10 1a507265 76696f75 7350726f 74656374 696f6e44 61746145 74616708 10018001 09800080 02800080 10800080 0e800c08 800b8009 80118000 80008000 80008000 80000008 0011001a 0023002d 00320037 004c0052 005e0065 006c0077 007e0080 00820084 00ab00b2 00bb00c5 00c700c9 00cb00d8 00eb00f0 00fb0104 01130116 0125012e 0133013e 0141014c 01510159 01620164 01690170 01730178 01810183 018a018c 018e0190 01a301aa 01ac01ae 01b001c3 01cc01d5 01e70214 021e0229 02340242 024e0257 02700275 028702a2 02b802cc 02d802e4 02f70306 03200328 033d0352 036f0370 03720374 03750377 0379037b 037d037f 03810383 03840386 0388038a 038c038e 03900392 03940000 00000000 02010000 00000000 00660000 00000000 00000000 00000000 0396> type NSConcreteMutableData

**Comment by evermeer** [comment]
Ah, I already was hesitating if I should do a warning. I already removed it but thought I could wait making a new version for it. I guess I should still have created a new version for it. 

For CloudKit it definitely can be ignored. The one from your initial commit states that it's for type CKRecordID. That one I was expecting. The second for type NSConcreteMutableData is strange. I don't know where that comes from.

Some background: Until 3 weeks ago I was using recursive reflection only if a property was of type EVObject or an array of that. Then I decided to also support NSObject for nested objects. There is where it went wrong for the CKRecordID because that one is a NSObject, but it can't be recreated based on a dictionary of its properties. That was the bug with the RecordID that you noticed. So this weekend I reverted that logic and replaced it for that warning. 

I think I will make it a requirement that properties for objects that should be processed should have EVObject as its base object. Using NSObject would almost be impossible to support since I can't know for sure if it's a special type like CKRecordID.
